src/Nethermind/Nethermind.TxPool/TxBroadcaster.cs

```diff
@@ -134,38 +134,48 @@ public void BroadcastPersistentTxs()
 
             if (_txPoolConfig.PeerNotificationThreshold > 0)
             {
-                IList<Transaction> persistentTxsToSend = GetPersistentTxsToSend();
+                (IList<Transaction>? transactionsToSend, IList<Transaction>? hashesToSend) = GetPersistentTxsToSend();
 
-                if (persistentTxsToSend.Count > 0)
+                if (transactionsToSend is not null)
                 {
-                    if (_logger.IsDebug) _logger.Debug($"Broadcasting {persistentTxsToSend.Count} persistent transactions to all peers.");
+                    if (_logger.IsDebug) _logger.Debug($"Broadcasting {transactionsToSend.Count} persistent transactions to all peers.");
 
                     foreach ((_, ITxPoolPeer peer) in _peers)
                     {
-                        Notify(peer, persistentTxsToSend, true);
+                        Notify(peer, transactionsToSend, true);
                     }
                 }
                 else
                 {
                     if (_logger.IsDebug) _logger.Debug($"There are currently no transactions able to broadcast.");
                 }
+
+                if (hashesToSend is not null)
+                {
+                    if (_logger.IsDebug) _logger.Debug($"Announcing {hashesToSend.Count} hashes of persistent transactions to all peers.");
+
+                    foreach ((_, ITxPoolPeer peer) in _peers)
+                    {
+                        Notify(peer, hashesToSend, false);
+                    }
+                }
             }
             else
             {
                 if (_logger.IsDebug) _logger.Debug($"PeerNotificationThreshold is not a positive value: {_txPoolConfig.PeerNotificationThreshold}. Skipping broadcasting persistent transactions.");
             }
         }
 
-        internal IList<Transaction> GetPersistentTxsToSend()
+        internal (IList<Transaction>? TransactionsToSend, IList<Transaction>? HashesToSend) GetPersistentTxsToSend()
         {
             // PeerNotificationThreshold is a declared in config max percent of transactions in persistent broadcast,
             // which will be sent after processing of every block. numberOfPersistentTxsToBroadcast is equal to
             // PeerNotificationThreshold multiplication by number of transactions in persistent broadcast, rounded up.
             int numberOfPersistentTxsToBroadcast =
-                Math.Min(_txPoolConfig.PeerNotificationThreshold * _persistentTxs.Count / 100 + 1,
-                    _persistentTxs.Count);
+                Math.Min(_txPoolConfig.PeerNotificationThreshold * _persistentTxs.Count / 100 + 1, _persistentTxs.Count);
 
-            List<Transaction> persistentTxsToSend = new(numberOfPersistentTxsToBroadcast);
+            List<Transaction>? persistentTxsToSend = null;
+            List<Transaction>? persistentHashesToSend = null;
```

Comment:
Can we use `ArrayPoolList's`? I think we could.

Reply:
Then we would need to declare it one method higher (in `BroadcastPersistentTxs()`) and pass both to `GetPersistentTxsToSend()` - otherwise it will be disposed too quick. We are calculating `numberOfPersistentTxsToBroadcast` (so max size of collections) in `GetPersistentTxsToSend()` - we would need to move it higher as well. It would be a bit ugly. And taking into consideration that
1) this method is called once per block
2) if there are no persistent txs (as in most of our nodes) right now we are not allocating anything - lists are `nulls` until there is an item eligible to add
3) using `ArrayPoolList` would be called no matter if we have eligible items

Actually it would be good to add basic check `_persistentTxs.Count == 0` in the beginning and quick return if collection is empty. And then using `ArrayPoolList` would have more sense. But then most of the logic from `GetPersistentTxsToSend()` would be moved higher, to `BroadcastPersistentTxs()`, so we could move everything and drop `GetPersistentTxsToSend()` at all. And we will still need to do `using ArrayPoolList` for both collections even if one would not have any items

I will add quick return in case of empty collection - it's a good idea. And about `ArrayPoolList's` - if you want I can do further refactor, but I'm not convinced to that idea

Reply:
Hm... Then instead of using ArrayPoolList, maybe we can have a List<> instance we can reuse? As this is single-threaded if I understand it correctly.